src/core/hle/service/boss/boss.cpp

```diff
@@ -185,6 +190,167 @@ void Module::Interface::GetStepIdList(Kernel::HLERequestContext& ctx) {
 
     LOG_WARNING(Service_BOSS, "(STUBBED) size={:#010X}", size);
 }
+auto Module::Interface::GetBossDataDir() {
+    u64 extdata_id = 0;
+    Core::System::GetInstance().GetAppLoader().ReadExtdataId(extdata_id);
+
+    const u32 high = static_cast<u32>(extdata_id >> 32);
+    const u32 low = static_cast<u32>(extdata_id & 0xFFFFFFFF);
+
+    return FileSys::ConstructExtDataBinaryPath(1, high, low);
+}
+std::vector<NsDataEntry> Module::Interface::GetNsDataEntries(u32 max_entries) {
+    std::vector<NsDataEntry> ns_data;
+    u32 entry_count = 0;
+    u32 files_to_read = 100;
+    FileSys::Entry boss_files[100];
+
+    entry_count = GetBossExtDataFiles(files_to_read, boss_files);
+
+    if (entry_count > max_entries) {
+        LOG_WARNING(Service_BOSS, "Number of output entries has exceeded maximum");
+        entry_count = max_entries;
+    }
+
+    for (u32 i = 0; i < entry_count; i++) {
+        if (boss_files[i].is_directory || boss_files[i].file_size < 52) {
+            LOG_WARNING(Service_BOSS, "Directory or too-short file in spotpass extdata");
+            continue;
+        }
+
+        NsDataEntry entry;
+        std::string filename{Common::UTF16ToUTF8(boss_files[i].filename)};
+        filename = "/" + filename;
+        FileSys::Path file_path = filename.c_str();
+        LOG_DEBUG(Service_BOSS, "Spotpass filename={}", filename);
+        entry.filename = filename;
+
+        FileSys::Mode mode{};
+        mode.read_flag.Assign(1);
+        FileSys::ArchiveFactory_ExtSaveData extdata_archive_factory(
+            FileUtil::GetUserPath(FileUtil::UserPath::SDMCDir), false);
+        FileSys::Path boss_path{GetBossDataDir()};
+        auto archive_result = extdata_archive_factory.OpenSpotpass(boss_path, 0);
+
+        if (archive_result.Succeeded()) {
```

Comment:
```suggestion
        if (!archive_result.Succeeded()) {
            LOG_WARNING(Service_BOSS, "Extdata opening failed");
            return;
        }
```
same with the other check below
it also seems like the extdata opening can happen outside the loop.